src/code/views/weather-styler.tsx

```diff
@@ -23,9 +23,15 @@ export function precipDiv(station?:IWeatherStation|null) {
   const raining = station && station.precipitation;
```

Comment:
There's a bit of redundancy here between `raining` and the precipitation handling below. Consider
```typescript
const precipitation = station && station.precipitation;
```
and then use precipitation here and down below.